main.py

```diff
@@ -194,6 +172,58 @@ def get_confirmation_link(self, row_id):
         url = "http://epscheduleapp.appspot.com/confirm/" + encoded_row_id
         return url
 
+class RegisterHandler (BaseHandler):
+    def post(self):
+        email = self.request.get('email')
+        password = self.request.get('password')
+
+        if email[-17:] != "@eastsideprep.org":
+            self.response.write(json.dumps(ERR_SIGNUP_EMAIL_NOT_EPS))
+            return
+
+        if not self.check_signed_up(email):
+            self.response.write(json.dumps(ERR_EMAIL_ALREADY_REGISTERED))
+            return
+
+        self.response.write(json.dumps(REGISTER_SUCCESS))
+        hashed = bcrypt.hashpw(password, bcrypt.gensalt(1))
+        user_obj = User(email = email, password = hashed, verified = False)
+        user_obj.join_date = datetime.datetime.now()
+        db.put(user_obj)
+        row_id = str(user_obj.key().id())
+        logging.info("row id = " + row_id)
+        self.send_confirmation_email(email, row_id)
+
+ERR_NO_ACCOUNT_TO_SEND = {
```

Comment:
I would just return a generic error here with no special button. This should never  happen.